shared/src/business/useCases/trialSessions/saveCalendarNoteInteractor.test.js

```diff
@@ -0,0 +1,185 @@
+      ...MOCK_CASE_WITH_TRIAL_SESSION,
+    };
+
+    applicationContext.getUniqueId.mockReturnValue(
+      '8675309b-18d0-43ec-bafb-654e83405411',
+    );
+
+    await saveCalendarNoteInteractor({
+      applicationContext,
+      calendarNote: 'whatever',
+      docketNumber: mockCase.docketNumber,
+      trialSessionId: MOCK_CASE_WITH_TRIAL_SESSION.trialSessionId,
+    });
+
+    expect(
+      applicationContext.getPersistenceGateway().getTrialSessionById,
+    ).toHaveBeenCalled();
+    expect(
+      applicationContext.getPersistenceGateway().getTrialSessionById.mock
+        .calls[0][0],
+    ).toMatchObject({
+      trialSessionId: MOCK_CASE_WITH_TRIAL_SESSION.trialSessionId,
+    });
+  });
+
+  it('successfully update the trial session with calendarNotes', async () => {
+    mockTrialSession.caseOrder = [
+      {
+        calendarNotes: 'this is not a calendar note',
+        docketNumber: mockCase.docketNumber,
+      },
+      {
+        calendarNotes: 'this is also not a calendar note',
+        docketNumber: '123-21',
+      },
+    ];
+
+    mockCase.trialSessionId = mockTrialSession.trialSessionId;
+
+    const result = await saveCalendarNoteInteractor({
+      applicationContext,
+      calendarNote: 'this is a calendarNote',
+      docketNumber: mockCase.docketNumber,
+      trialSessionId: mockTrialSession.trialSessionId,
+    });
+
+    expect(result.trialSessionId).toEqual(mockTrialSession.trialSessionId);
+    expect(
+      applicationContext.getPersistenceGateway().updateTrialSession,
+    ).toHaveBeenCalled();
+    expect(
+      applicationContext.getPersistenceGateway().updateTrialSession.mock
+        .calls[0][0].trialSessionToUpdate.caseOrder,
+    ).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          calendarNotes: 'this is a calendarNote',
+          docketNumber: mockCase.docketNumber,
+        }),
+        expect.objectContaining({
+          calendarNotes: 'this is also not a calendar note',
+          docketNumber: '123-21',
+        }),
+      ]),
+    );
+  });
+
+  it('does not update the case hearing record if the given trial session is not a hearing on the case', async () => {
+    await saveCalendarNoteInteractor({
+      applicationContext,
+      calendarNote: 'this is a calendarNote',
+      docketNumber: mockCase.docketNumber,
+      trialSessionId: mockTrialSession.trialSessionId,
+    });
+
+    expect(
+      applicationContext.getPersistenceGateway().getCaseByDocketNumber,
+    ).toHaveBeenCalled();
+    expect(
+      applicationContext.getPersistenceGateway().updateCaseHearing,
+    ).not.toHaveBeenCalled();
+  });
+
+  it('updates the case hearing record if the given trial session is a hearing on the case', async () => {
+    mockCase = {
+      ...MOCK_CASE,
+      hearings: [
+        {
+          ...MOCK_TRIAL,
+          caseOrder: [
+            {
+              calendarNotes: 'just a hearing note',
+              docketNumber: mockCase.docketNumber,
```

Comment:
mayhaps this ought to be `MOCK_CASE.docketNumber`

Reply:
Updated!